src/link/persisted-queries/index.ts

```diff
@@ -172,115 +175,112 @@ export const createPersistedQueryLink = (
-                  result && (result.errors as GraphQLFormattedError[]);
+                const result = JSON.parse(networkError.bodyText) as
+                  | FormattedExecutionResult
+                  | undefined;
 
-                if (isNonEmptyArray(networkErrors)) {
-                  graphQLErrors.push(...networkErrors);
-                }
+                graphQLErrors = result?.errors;
               } catch {}
             }
 
-            const disablePayload: ErrorResponse = {
-              response,
-              networkError,
-              operation,
-              graphQLErrors:
-                isNonEmptyArray(graphQLErrors) ? graphQLErrors : void 0,
-              meta: processErrors(graphQLErrors),
-            };
-
-            // if the server doesn't support persisted queries, don't try anymore
-            supportsPersistedQueries = !disable(disablePayload);
-            if (!supportsPersistedQueries) {
-              delete operation.extensions.persistedQuery;
-              // clear hashes from cache, we don't need them anymore
-              resetHashCache();
-            }
-
-            // if its not found, we can try it again, otherwise just report the error
-            if (retry(disablePayload)) {
-              // need to recall the link chain
-              if (subscription) subscription.unsubscribe();
-              // actually send the query this time
-              operation.setContext({
-                http: {
-                  includeQuery: true,
-                  ...(supportsPersistedQueries ?
-                    { includeExtensions: true }
-                  : {}),
-                },
-                fetchOptions: {
-                  // Since we're including the full query, which may be
-                  // large, we should send it in the body of a POST request.
-                  // See issue #7456.
-                  method: "POST",
-                },
-              });
-              if (setFetchOptions) {
-                operation.setContext({ fetchOptions: originalFetchOptions });
-              }
-              subscription = forward(operation).subscribe(handler);
-
-              return;
-            }
-          }
-          cb();
-        };
-        const handler = {
-          next: (response: FormattedExecutionResult) => {
-            maybeRetry({ response }, () => observer.next!(response));
-          },
-          error: (networkError: ServerError) => {
-            maybeRetry({ networkError }, () => observer.error!(networkError));
+            handleRetry(
+              {
+                error:
+                  isNonEmptyArray(graphQLErrors) ?
+                    new CombinedGraphQLErrors({ errors: graphQLErrors })
+                  : networkError,
+                operation,
+                meta: processErrors(graphQLErrors),
+              },
+              () => observer.error(networkError)
+            );
```

Comment:
```suggestion
          error: (incomingError) => {
            let networkError = toErrorLike(incomingError);

            // This is persisted-query specific (see #9410) and deviates from the
            // GraphQL-over-HTTP spec for application/json responses.
            // This is intentional.
            if (ServerError.is(networkError) && networkError.bodyText) {
              try {
                const result = JSON.parse(networkError.bodyText) as
                  | FormattedExecutionResult
                  | undefined;
                if (isNonEmptyArray(result?.errors)) {
                  networkError = new CombinedGraphQLErrors(result);
                }
              } catch {}
            }

            handleRetry(
              {
                error: networkError,
                operation,
                meta: processErrors(
                  CombinedGraphQLErrors.is(networkError) ?
                    networkError.errors
                  : []
                ),
              },
              () => observer.error(incomingError)
            );
```

What do you think about this change?